tests/test_cache_lock_file_perf.py

```diff
@@ -0,0 +1,110 @@
+import multiprocessing
+import threading
+import time
+from msal_extensions import FilePersistence, CrossPlatLock
+
+
+def validate_thread_locking_file(num):
+    with open("C:/Users/abpati/Desktop/test/lockfilelog.txt") as handle:
+        data = handle.read()
+    time_list = []
+    for line in data.split("\n"):
+        start, end = line.split("-")
+        time_list.append([start, end])
+    if len(time_list) != num:
+        print("Threads count dont match ")
+        return
+    time_list.sort()
+    prev = None
+    for interval in time_list:
+        start, end = interval
+        if start > end:
+            print("Intervals start bigger than end")
+            return
+        if prev:
+            if start < prev:
+                print("overlapping")
+        prev = start
+    print("success")
+    return
+
+
+def validate_result_in_cache(num):
+    with open("C:/Users/abpati/Desktop/test/msal.cache") as handle:
+        data = handle.read()
+    prev_tag= None
+    prev_proc_id = None
+    count = 0
+    for line in data.split(("\n")):
+        tag, proc_id = line.split(" ")
+        if tag == '<':
+            if '<' == prev_tag:
+                print("Failed")
+                return
+        elif tag == '>':
+            count+=1
+            if '<' != prev_tag or (not prev_proc_id == proc_id):
+                print("Failed")
+                return
+        else:
+            print("Unexpected Token")
+        prev_proc_id = proc_id
+        prev_tag = tag
+    if ">" != prev_tag:
+        print("Failed")
+        return
+    if count != num:
+        print("Failed")
+        return
+    print("sucess")
+
+
+def do_something(tid):
+    lock_intervals_file_path = "C:/Users/abpati/Desktop/test/lockfilelog.txt"
+    lock_file_path = "C:/Users/abpati/Desktop/test/msal.cache.lockfile"
+    cache_accessor = FilePersistence("C:/Users/abpati/Desktop/test/msal.cache")
+    with CrossPlatLock(lock_file_path, lock_intervals_file_path):
+        thr = str(tid)
+        data = cache_accessor.load()
+        if data is None:
+            data = ""
+        data += "< " + thr + "\n"
+        time.sleep(0.1)
```

Comment:
Did you notice different behvior with 1s instead of 0.1s here ? I think Java and .NET used 1s to make sure we get higher concurrency..

Reply:
This parameter should become a parameter of this method, so that we can better customize the configuration in test automation.